autoform-hooks.js

```diff
@@ -1,21 +1,30 @@
 // Manages all hooks, supporting append/replace, get
 
 Hooks = {
-  form: {},
-  global: {
+  form: {}
+};
+
+/**
+ * @return {String} The names of all supported hooks, excluding "before" and "after".
+ */
+function getHookNames() {
+  return ['formToDoc', 'formToModifier', 'docToForm', 'onSubmit', 'onSuccess',
+      'onError', 'beginSubmit', 'endSubmit'];
+};
```

Comment:
This doesn't need to be a function. You can do `var hookNames = []` and `_.each(hookNames, ...)`.